Update Developer Guide

bryanhce opened the pull request:
Update target user profile, value proposition, use cases, non-functional requirements and glossary.

bryanhce committed:
Update Developer Guide

Update target user profile, value proposition, use cases, non-functional
requirements and glossary.

jehousoh commented on docs/DeveloperGuide.md:
```diff
@@ -283,16 +312,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  
+    Use case resumes at step 2.
+  
+**Use case: Delete a guest**
```
Should we shift the use cases to be ordered based on what function comes first like our UG? ie Add -> Delete -> List -> Exit

bryanhce replied:
Yes, we should follow that order. Will change.

jehousoh reviewed (changes requested):
LGTM other than the order of the use cases!

bryanhce committed:
Reorder use cases

Reorder to follow the same order as the user guide of use case about
add, use case about delete, use case about list, use case about exit

jehousoh reviewed (approved):
lgtm!
